src/test/kotlin/org/rust/lang/core/completion/RsCompletionTestFixture.kt

```diff
@@ -5,125 +5,15 @@
 
 package org.rust.lang.core.completion
 
-import com.intellij.codeInsight.lookup.LookupElement
-import com.intellij.openapi.project.Project
-import com.intellij.openapi.vfs.VirtualFileFilter
-import com.intellij.psi.impl.PsiManagerEx
 import com.intellij.testFramework.fixtures.CodeInsightTestFixture
-import com.intellij.testFramework.fixtures.impl.BaseFixture
-import org.intellij.lang.annotations.Language
-import org.rust.*
+import org.rust.InlineFile
 
 class RsCompletionTestFixture(
-    private val myFixture: CodeInsightTestFixture,
+    fixture: CodeInsightTestFixture,
     private val defaultFileName: String = "main.rs"
-) : BaseFixture() {
+) : RsCompletionTestFixtureBase<String>(fixture) {
 
-    private val project: Project get() = myFixture.project
-
-    fun executeSoloCompletion() {
-        val variants = myFixture.completeBasic()
-
-        if (variants != null) {
-            if (variants.size == 1) {
-                // for cases like `frob/*caret*/nicate()`,
-                // completion won't be selected automatically.
-                myFixture.type('\n')
-                return
-            }
-            fun LookupElement.debug(): String = "$lookupString ($psiElement)"
-            error("Expected a single completion, but got ${variants.size}\n"
-                + variants.joinToString("\n") { it.debug() })
-        }
-    }
-
-    fun doFirstCompletion(before: String, after: String) {
-        check(hasCaretMarker(before) && hasCaretMarker(after)) {
```

Comment:
Maybe keep `hasCaretMarker(after)` check here and in `doSingleCompletion`? Or add such check to `checkByText`

Reply:
This check is performed in `org.rust.InlineFile#withCaret` that we call in `org.rust.lang.core.completion.RsCompletionTestFixture#prepare`

Reply:
Check in `org.rust.InlineFile#withCaret` is performed for `before`, and for `after` there is no such check if I understand correctly

Reply:
You are right
Thanks!